kore/src/Kore/Step/Strategy.hs

```diff
@@ -414,6 +421,8 @@ constructExecutionGraph transit instrs0 searchOrder0 config0 = do
       | []              <- instrs = unfoldWorker rest searchOrder
       | instr : instrs' <- instrs
       = Profile.executionQueueLength (Seq.length rest) $ do
+        when ((not . withinLimit depthLimit . fromIntegral . Seq.length) rest)
```

Comment:
Let's factor out this error into a helper function. We should also `unparse` the branches for the error message.